python/ray/air/checkpoint.py

```diff
@@ -238,20 +268,34 @@ def from_dict(cls, data: dict) -> "Checkpoint":
         Returns:
             Checkpoint: checkpoint object.
         """
-        return cls(data_dict=data)
+        state = {}
+        if isinstance(data, CheckpointDict):
```

Comment:
hmm this seems a bit weird to me? if `data` is `bytes`, then how can it be an instance of `CheckpointDict`?